sdks/python/apache_beam/examples/snippets/snippets_test.py

```diff
@@ -693,6 +695,64 @@ def test_model_co_group_by_key_tuple(self):
     expect = ['a; a@example.com; x4312', 'b; b@example.com; x8452']
     self.assertEqual(expect, self.get_output(result_path))
 
+  def test_model_use_and_query_metrics(self):
+    """DebuggingWordCount example snippets."""
+
+    import re
+    import apache_beam as beam
+
+    p = TestPipeline()  # Use TestPipeline for testing.
+    words = p | beam.Create(['metrics', 'usage', 'Flourish', 'stomach',
+                             'a', 'longword10'])
+
+    # [START metrics_usage_example]
+    class FilterTextFn(beam.DoFn):
+      """A DoFn that filters for a specific key based on a regex."""
+
+      def __init__(self, pattern):
+        self.pattern = pattern
+        # A custom metric can track values in your pipeline as it runs. Create
+        # custom metrics to count matched words, and know the distribution of
+        # word lengths in the input PCollection.
+        self.word_len_dist = Metrics.distribution(self.__class__,
+                                                  'word_len_dist')
+        self.matched_words = Metrics.counter(self.__class__, 'matched_words')
+
+      def process(self, element):
+        word = element
+        self.word_len_dist.update(len(word))
+        if re.match(self.pattern, word):
+          self.matched_words.inc()
+          yield element
```

Comment:
Nit: this is equal to the size of filtered_words; perhaps let the counter be omitted/dropped words?